scitt/check_operation_status.py

```diff
@@ -41,20 +42,32 @@ def get_operation_status(operation_id: str, headers: dict) -> dict:
     return response.json()
 
 
-def poll_operation_status(operation_id: str, headers: dict) -> str:
+def poll_operation_status(
+    operation_id: str, headers: dict, logger: logging.Logger
+) -> str:
     """
     polls for the operation status to be 'succeeded'.
     """
 
     poll_attempts: int = int(POLL_TIMEOUT / POLL_INTERVAL)
 
+    logger.info("starting to poll for operation status 'succeeded'")
+
     for _ in range(poll_attempts):
-        operation_status = get_operation_status(operation_id, headers)
 
-        # pylint: disable=fixme
-        # TODO: ensure get_operation_status handles error cases from the rest request
-        if "status" in operation_status and operation_status["status"] == "succeeded":
-            return operation_status["entryID"]
+        try:
+            operation_status = get_operation_status(operation_id, headers)
+
+            # pylint: disable=fixme
+            # TODO: ensure get_operation_status handles error cases from the rest request
+            if (
+                "status" in operation_status
+                and operation_status["status"] == "succeeded"
+            ):
+                return operation_status["entryID"]
+
+        except requests.HTTPError as e:
+            logger.debug(f"failed getting operation status, error: {e}")
```

Comment:
its inefficient to use f strings - this ode will format the string regardles of whether the logging level is set. Use old-style C format such as %s instead.

https://docs.python.org/3/howto/logging.html#logging-variable-data

Reply:
good suggestion :) done!